### PR TITLE
Enable more libsolv features and rework spec distro conditionals

### PR DESCRIPTION
**Pull request by Conan-Kudo**
- Enable bzip2 and xz/lzma compression support
- Enable complex/rich dependencies on distributions with RPM 4.13+
- Simplified CentOS/RHEL conditionals
- Added Mageia conditionals
- Fixed a few spec portability issues

### CODE REVIEW

**Comment by Conan-Kudo** [comment]
@mlschroe Ping?

**Comment by Conan-Kudo** [comment]
@mlschroe I've disabled all the new functionality for SLES 12 and older. It'll enable itself as appropriate.

**Comment by Conan-Kudo** [comment]
@mlschroe Ping?

**Comment by Conan-Kudo** [comment]
@mlschroe Can you please merge this? Without this merged, I can't proceed with submitting DNF into openSUSE, as libdnf will not link against libsolv as it's currently configured.

File: package/libsolv.changes [hunk 1]
```diff
@@ -1,4 +1,10 @@
 -------------------------------------------------------------------
+Sun May 28 13:32:15 UTC 2017 - ngompa13@gmail.com
+
+- Enable complex/rich dependencies for CentOS/RHEL 7, matching how
```
Comment (ignatenkobrain) [inline package/libsolv.changes]: RPM on RHEL7 doesn't support rich dependencies

Comment (Conan-Kudo) [inline package/libsolv.changes]: But the feature is enabled on libsolv in CentOS 7. I checked [the libsolv spec](https://git.centos.org/blob/rpms!libsolv/662e5a792ec1643ca54c2d1ff45c5a4eaca3ed15/SPECS!libsolv.spec#L26) to be sure.

Comment (ignatenkobrain) [inline package/libsolv.changes]: that seems wrong, feel free to report a bug ;)

Comment (Conan-Kudo) [inline package/libsolv.changes]: @ignatenkobrain Done (see [rh#1456272](https://bugzilla.redhat.com/show_bug.cgi?id=1456272))

**Pull request merged**